benchmarks/python/test_rope.py

```diff
@@ -176,3 +176,1568 @@ def test_rope_benchmark(
+            ((4096, 128), torch.bfloat16),
+            ((4096, 128), torch.bfloat16),
+        ],
+    ),
+    "llama_2_7b_hf_rope_bwd": (
+        llama_2_7b_hf_rope_bwd,
+        [
+            ((4096, 128), torch.bfloat16),
+            ((2, 32, 4096, 128), torch.bfloat16),
+            ((2, 32, 4096, 128), torch.bfloat16),
+            ((4096, 128), torch.bfloat16),
+        ],
+    ),
+    "llama_3_8B_rope_fwd": (
+        llama_3_8B_rope_fwd,
+        [
+            ((2, 8192, 6144), torch.bfloat16),
+            ((8192, 128), torch.bfloat16),
+            ((8192, 128), torch.bfloat16),
+        ],
+    ),
+    "llama_3_8B_rope_bwd": (
+        llama_3_8B_rope_bwd,
+        [
+            ((8192, 128), torch.bfloat16),
+            ((2, 32, 8192, 128), torch.bfloat16),
+            ((2, 32, 8192, 128), torch.bfloat16),
+            ((8192, 128), torch.bfloat16),
+        ],
+    ),
+    "hf_qwen2_rope_fwd": (
+        hf_qwen2_rope_fwd,
+        [
+            ((1, 32768, 3584), torch.bfloat16),
+            ((1, 32768, 512), torch.bfloat16),
+            ((1, 32768, 512), torch.bfloat16),
+            ((1, 32768, 128), torch.bfloat16),
+            ((1, 32768, 128), torch.bfloat16),
+        ],
+    ),
+    "hf_qwen2_rope_bwd": (
+        hf_qwen2_rope_bwd,
+        [
+            ((1, 28, 32768, 128), torch.bfloat16),
+            ((1, 32768, 3584), torch.bfloat16),
+            ((1, 32768, 512), torch.bfloat16),
+            ((1, 32768, 128), torch.bfloat16),
+            ((1, 28, 32768, 128), torch.bfloat16),
+            ((1, 28, 32768, 128), torch.bfloat16),
+            ((1, 32768, 128), torch.bfloat16),
+        ],
+    ),
+    "hf_phi3_rope_fwd": (
+        hf_phi3_rope_fwd,
+        [
+            ((2, 4096, 9216), torch.bfloat16),
+            ((48,), torch.bfloat16),
+            ((1, 4096), torch.int64),
+        ],
+    ),
+    "hf_phi3_rope_bwd": (
+        hf_phi3_rope_bwd,
+        [
+            ((2, 32, 4096, 96), torch.bfloat16),
+            ((2, 32, 4096, 96), (0, 0, 96, 1), torch.bfloat16),
+            ((2, 32, 4096, 96), torch.bfloat16),
+            ((2, 32, 4096, 96), (0, 0, 96, 1), torch.bfloat16),
+            ((2, 32, 4096, 96), torch.bfloat16),
+        ],
+    ),
+    "hf_mistral_nemo_rope_fwd": (
+        hf_mistral_nemo_rope_fwd,
+        [
+            ((1, 128000, 4096), torch.bfloat16),
+            ((1, 128000, 1024), torch.bfloat16),
+            ((1, 128000, 1024), torch.bfloat16),
+            ((64,), torch.bfloat16),
+            ((1, 128000), torch.int64),
+        ],
+    ),
+    "hf_mistral_nemo_rope_bwd": (
+        hf_mistral_nemo_rope_bwd,
+        [
+            ((1, 32, 128000, 128), torch.bfloat16),
+            ((1, 1, 128000, 128), torch.bfloat16),
+            ((1, 32, 128000, 128), torch.bfloat16),
+            ((1, 32, 128000, 128), torch.bfloat16),
+            ((1, 1, 128000, 128), torch.bfloat16),
+        ],
+    ),
+}
+
+
+@pytest.mark.parametrize(
```

Comment:
This is the only part that's worth reviewing.

code above were directly dumped from Kevin's rope example script. (Note that I have to update the script with `nv_enable_matmul` in thunder.jit, otherwise we are seeing segmentation at nvfuser definition level)